libs/core/src/lib/button/button.component.html

```diff
@@ -1,15 +1,19 @@
-<ng-container *ngIf="glyphPosition === 'before' && glyph">
-    <fd-icon [glyph]="glyph"></fd-icon>
+<ng-container *ngIf="glyphPosition === 'before'">
+    <ng-container [ngTemplateOutlet]="iconTemplate"></ng-container>
 </ng-container>
 
 <span class="fd-button__text" *ngIf="label">
     {{ label }}
 </span>
 
-<ng-content></ng-content>
-
-<ng-container *ngIf="glyphPosition === 'after' && glyph">
-    <fd-icon [glyph]="glyph"></fd-icon>
+<ng-container *ngIf="glyphPosition === 'after'">
+    <ng-container [ngTemplateOutlet]="iconTemplate"></ng-container>
 </ng-container>
 
 <fd-icon glyph="slim-arrow-down" *ngIf="fdMenu"></fd-icon>
+
+<ng-template #iconTemplate>
+    <fd-icon *ngIf="glyph" [glyph]="glyph"></fd-icon>
+
+    <ng-content></ng-content>
```

Comment:
Maybe keep original ng-content, and use this one with [select]?
There might be cases when button is used like so:
```
<button fd-button>
   Some inner text
   <fd-icon...></fd-icon>
</buton>```

Reply:
Adding `[select]` is fine, though it adds one more detail to think of. 

Your example won't be affected with current changes, but still no any new details added.

Reply:
`glyphPosition` is `before` by default, so we show projected content even if nothing (like in your example) specified, neither glyph, label or glyphPosition.